compiler/seminst.nim

```diff
@@ -335,7 +335,12 @@ proc generateInstance(c: PContext, fn: PSym, pt: TIdTable,
   while not isTopLevel(c): c.currentScope = c.currentScope.parent
   result = copySym(fn, c.idgen)
   incl(result.flags, sfFromGeneric)
-  result.owner = fn
+  result.instantiatedFrom = fn
+  if sfGlobal in result.flags and 
+    (c.config.symbolFiles != disabledSf or "nimInstantiatedFromUsage" in c.config.symbols):
+    result.owner = c.module 
```

Comment:
The owner should always be `c.module`.

Reply:
Im afraid that would break generic `hooks=` as it would move them away from the module where the type is defined

Reply:
It should not break generic hooks.

Reply:
I mean, that was the issue below where I undo the movement (arraymancer has an use of it where the tests fails). I can show you the case if you want. 

But the test is just the way to toggle it on/off. Remember that it breaks current libraries that rely on locals `passC`

Reply:
The error is: 
```Error: type bound operation `deallocTilesFinalizerWrapper` can be defined only in the same module with its type (Tiles:ObjectType)```

To sumarize: 
- Owners being the source generic still can be defined through generics where we need to revert them. (Not sure if trying to move the instance too would get rid of this issue?)

- Owners being the source generic needs to be retrocompatible with libs not using IC (maybe we can add a transition period?)